pkg/plugins/policies/core/xds/meshroute/clusters.go

```diff
@@ -117,16 +119,30 @@ func GenerateClusters(
 			}
 
 			resources = resources.Add(&core_xds.Resource{
-				Name:     clusterName,
-				Origin:   generator.OriginOutbound,
-				Resource: edsCluster,
+				Name:           clusterName,
+				Origin:         generator.OriginOutbound,
+				Resource:       edsCluster,
+				ResourceOrigin: createOwner(service.BackendRef(), meshCtx),
+				Protocol:       protocol,
 			})
 		}
 	}
 
 	return resources, nil
 }
 
+func createOwner(ref common_api.BackendRef, meshCtx xds_context.MeshContext) *core_rules.UniqueResourceIdentifier {
+	if ref.Kind == common_api.MeshService && ref.ReferencesRealObject() {
+		ms := meshCtx.MeshServiceByName[ref.Name]
```

Comment:
At this moment, `meshCtx.MeshServiceByName` contains `ms.GetMeta().GetName()` as keys which means on k8s it's equal to `name.namespace`. And `ref.Name` is probably just a name right? 

I think `meshCtx.MeshServiceByName` map should contain `ResourceIdentifier` as keys. And when you want to lookup `ref` in this map you'll need to call `TargetRefToResourceIdentifier(meta, ref)`

Reply:
That's odd, does it work on k8s?

Reply:
Yes, I tested it on k8s, and it worked in e2e test for mesh timeout and resource rules

Reply:
backendRef for real MeshService have `name.namespace` as name, that's why this work